h2o-genmodel/src/main/java/hex/genmodel/utils/DistributionFamily.java

```diff
@@ -8,6 +8,7 @@
 public enum DistributionFamily {
   AUTO,  // model-specific behavior
   bernoulli,
+  binomial,
```

Comment:
why were these new distribution types not needed for the algo impl, but are needed for mojo?

Reply:
@honzasterba:  For GLM, the families are defined as Gaussian, Binomial, Multinomial,....

For some reason, the other H2O toolboxes use Bernoulli for Binomial.  That is why I added the binomial to the DistributionFamily to accommodate both ways.

Reply:
Also, thanks for reminding me about this.  I left out one family that are binomial as well, namely, fractionalbinomial.  Thanks.

Reply:
@wendycwong, are you sure, you are using correct `DistributionFamily` enum and  `LinkFunctionType` enum? Because the GLM uses its own implementation of these enums. Other implementations use these enums. That is why binomial is missing here. The GLM really needs to refactor these parts of code to be consistent with other algos. So if it possible for GAM to use also Bernoulli, it is better than add the same distribution with the different name into this enum.

Reply:
@wendycwong, I am sorry, but I can't see this issue was resolved. There are actually two possibilities:
1. use GLM Distribution enum
OR
2. use this enum but with Bernoulli distribution.

Add the same distribution whit a different name is confusing here.

Reply:
@maurever:  Yes, I am sure that the DistributionFamily and LinkFunctionType are correct for GLM.  GLM does not have its own implementation but it does have its own names.  In addition, GAM defers to GLM to perform link function evaluation:

  double evalLink(double val) {
    switch (_link_function) {
      case identity: return GenModel.GLM_identityInv(val);
      case logit: return GenModel.GLM_logitInv(val);
      case log: return GenModel.GLM_logInv(val);
      case inverse: return GenModel.GLM_inverseInv(val);
      case tweedie: return GenModel.GLM_tweedieInv(val, _tweedieLinkPower);
      default: throw new UnsupportedOperationException("Unexpected link function "+_link_function);
    }
  }

I am not going to change GLM for now.  Plus we cannot change GLM family now anyway in order to avoid breaking use code.  I can make GAM mojo recognize Bernoulli though here.